File: backend/app/api/schemas/mentor.py
"""
Mentor API Schemas
"""

from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime


class MentorStatsResponse(BaseModel):
    """Mentor statistics response"""
    id: str
    mentor_name: str
    total_sessions: int
    average_score: float
    last_session: Optional[datetime] = None


class MentorListResponse(BaseModel):
    """List of mentors response"""
    mentors: List[MentorStatsResponse]


class TopMentorResponse(BaseModel):
    """Top mentor response"""
    rank: int
    id: str
    mentor_name: str
    average_score: float
    total_sessions: int


class TopMentorsListResponse(BaseModel):
    """List of top mentors response"""
    top_mentors: List[TopMentorResponse]


class MentorDetailResponse(BaseModel):
    """Detailed mentor information"""
    mentor_id: str
    mentor_name: str
    total_sessions: int
    completed_sessions: int
    average_score: float
    highest_score: float
    lowest_score: float
    recent_analyses: List[dict]

File: frontendorg/src/types/mentor.ts
export interface MentorStats {
  id: string;
  mentor_name: string;
  total_sessions: number;
  average_score: number;
  last_session?: string;
}

export interface TopMentor {
  rank: number;
  id: string;
  mentor_name: string;
  average_score: number;
  total_sessions: number;
}

export interface MentorDetail {
  mentor_id: string;
  mentor_name: string;
  total_sessions: number;
  completed_sessions: number;
  average_score: number;
  highest_score: number;
  lowest_score: number;
  recent_analyses: any[];
}

export interface MentorListResponse {
  mentors: MentorStats[];
}

export interface TopMentorsResponse {
  top_mentors: TopMentor[];
}